### PR TITLE
Nice-to-haves and fixes for Makefile

### PR DESCRIPTION
**Pull request by JakubVanek**
Hi,

this PR adds a mix of installation-related fixes and mini-features to Makefile.
- the instalation code somehow stopped working, this is now fixed
- example now works again
- object files are in their own directory, so they do not pollute the code directories
- it now has dependency tracking
- it supports pretty-printing the progress with ` [CC] ` prefixes
- `.DEFAULT` is replaced with `all`

### CODE REVIEW

**Comment by JakubVanek** [comment]
I have created a wrapper layer around EV3-API written in Make; I've published it at https://github.com/JakubVanek/c4ev3-workspace. If you deem it useful and correct, I'd be happy to have it forked into the c4ev3 organization.

It supports a similar workspace-based model to what https://github.com/c4ev3/EV3-API/pull/37 was motivated by. It does not have automatic CLion support (CMake-only :/), but for terminal-based workflow it still makes the things easier.

File: Makefile [hunk 1]
```diff
@@ -1,46 +1,91 @@
...
+# define flags
+DEPFLAGS = -MT $@ -MMD -MP -MF $(DEPDIR)/$*.d
 override CFLAGS += -std=c99
 override CFLAGS += -fno-strict-aliasing -fwrapv
 override CFLAGS += -Wall -Wextra -Wpointer-sign -Wno-unused-parameter
 override CFLAGS += -D_GNU_SOURCE=1
 
-.DEFAULT: libev3api.a
+# logging control
+ifeq ($(VERBOSE),)
```
Comment (a3f) [inline Makefile]: Nitpick: `ifeq ($(VERBOSE)$(V),)` for the Autotools impaired..

**Comment by a3f** [comment]
Workspace repo looks good.  But maybe it would be better to have it in EV3-API itself as projects/make` or `examples/make`? If someone writes a CMake scripts using the pkgconfig, they can add it likewise as `projects/cmake` as well. Thoughts?

File: Makefile (hunk 1 above)
Comment (JakubVanek) [reply Makefile]: That's an elegant way of doing an OR, thanks! Fixed in https://github.com/c4ev3/EV3-API/pull/49/commits/e173fc611889534aa8f23592fe4870df96a77c05

**Comment by JakubVanek** [comment]
Yes, I agree. I have modified it slightly and included it in this PR.

**Pull request merged**